### PR TITLE
SDK-1151: Allow multiple attributes with the same name

### PR DESCRIPTION
**Pull request by davidgrayston**
> Ported from #97 

### Changed
- `Yoti\Profile\BaseProfile::getAttributesByName()` is now public
- `Yoti\ShareUrl\Policy\DynamicPolicyBuilder::withWantedAttribute()` now allows the same wanted attribute with different constraints

### CODE REVIEW

**Comment by echarrod** [comment]
Looking at https://github.com/getyoti/yoti-php-sdk/pull/97, it looks like the examples used to be in the `public` folder. Could you copy the changes in this PR which are in `public` to the current file locations, and we can see if there are any changes?

**Comment by LuborRod** [comment]
> 

I have already checked everything, the same logic. It will not be same code, because of in new examples we use blade from laravel for view.

File: tests/Profile/BaseProfileTest.php [hunk 1]
```diff
@@ -69,7 +69,6 @@ public function setup(): void
      * @covers ::__construct
      * @covers ::getProfileAttribute
      * @covers ::setAttributesMap
-     * @covers ::getAttributesByName
```
Comment (echarrod) [inline tests/Profile/BaseProfileTest.php:72]: Should this line have been removed?

**Review by echarrod — COMMENTED** [review]
I think we still have some of the changes in https://github.com/getyoti/yoti-php-sdk/pull/97 that we need to have in this PR too: 
- src/Yoti/Util/Profile/AttributeListConverter.php (convertToYotiAttributesList)
- other changes in src/Yoti/Entity/BaseProfile.php (attributesMap, attributesList)
- the accompanying tests for these

File: tests/Profile/BaseProfileTest.php (hunk 1 above)
Comment (LuborRod) [reply tests/Profile/BaseProfileTest.php:72]: 1. We have this method here - Yoti\Profile\Util\Attribute\AttributeListConverter and tests in Yoti\Test\Profile\Util\Attribute\AttributeListConverterTest.

2.We have these methods in another classes also, instead set we use construct method and convert everything before call constructor in class above.

3. All these methods are covered by tests, we have 100% )

4. Should be removed - David created special test for this method on 95 line.

**Review by echarrod — APPROVED** [review]
I see those changes I mentioned are already in development, so all good :+1:

**Pull request merged**